SQL/SQL (basic).sublime-syntax

```diff
@@ -1087,6 +1144,73 @@ contexts:
     - meta_content_scope: meta.table-alias-name.sql
     - include: immediately-pop
 
+  expect-user-name:
+    # prevent prototypes from inheriting syntaxes
+    - meta_include_prototype: false
+    - include: comments
+    - include: built-in-user-function-call
+    - match: (?=\S)
+      set: [user-name, user-identifier]
+
+  user-name:
+    - meta_include_prototype: false
+    - meta_content_scope: meta.username.sql
+    - include: immediately-pop
+
+  expect-user-name-declaration:
+    # prevent prototypes from inheriting syntaxes
+    - meta_include_prototype: false
+    - include: comments
+    - match: (?=\S)
+      set: [user-name-declaration, user-identifier]
+
+  user-name-declaration:
+    - meta_include_prototype: false
+    - meta_content_scope: entity.name.user.sql
+    - include: immediately-pop
+
+  user-identifier:
+    # https://mariadb.com/kb/en/grant/#account-names
+    - meta_include_prototype: false
+    - include: pop-on-top-level-reserved-word
+    - match: ''
+      set:
+        - identifier-host-accessor
+        - identifier-part
+
+  identifier-host-accessor:
+    - meta_include_prototype: false
+    - match: \s*(@)
+      captures:
+        1: punctuation.accessor.at.sql
+      set: identifier-host-part
+    - include: immediately-pop
+
+  identifier-host-part:
+    - meta_include_prototype: false
+    - match: \%
+      scope: constant.other.wildcard.percent.sql
+      pop: 1
+    - match: (')(%)(')
+      captures:
+        1: punctuation.definition.identifier.begin.sql
+        2: constant.other.wildcard.percent.sql
+        3: punctuation.definition.identifier.end.sql
+      pop: 1
+    - match: (")(%)(")
+      captures:
+        1: punctuation.definition.identifier.begin.sql
+        2: constant.other.wildcard.percent.sql
+        3: punctuation.definition.identifier.end.sql
+      pop: 1
+    - match: (`)(%)(`)
+      captures:
+        1: punctuation.definition.identifier.begin.sql
+        2: constant.other.wildcard.percent.sql
+        3: punctuation.definition.identifier.end.sql
+      pop: 1
+    - include: identifier-part
```

Comment:
I'm curious to understand why this belongs in the `basic` file and not in `MySQL`?

Reply:
Moved the `user-identifer` backtto MySQL, keeping simple `expect-user-name` only.